addons/account/models/account_reconcile_model.py

```diff
@@ -665,10 +665,13 @@ def _check_rule_propositions(self, statement_line, candidates):
                 total_residual += aml['aml_currency_id'] and aml['aml_amount_currency'] or aml['aml_balance']
             else:
                 total_residual += aml['aml_currency_id'] and aml['aml_amount_residual_currency'] or aml['aml_amount_residual']
+            total_currency = aml['aml_currency_id'] and self.env['res.currency'].browse(aml['aml_currency_id']) or self.company_id.currency_id
```

Comment:
Are you sure all `candidates` have the same currency?

Reply:
nice, I was testing with just a single candidate :)